src/servers/main.cc

```diff
@@ -212,64 +252,115 @@ SignalHandler(int signum)
 {
-  std::unique_ptr<nvidia::inferenceserver::GRPCServer> service;
   nvidia::inferenceserver::Status status =
-      nvidia::inferenceserver::GRPCServer::Create(server, grpc_port_, &service);
+      nvidia::inferenceserver::GRPCServer::Create(
+          server, port_map, &grpc_endpoint_services_);
   if (status.IsOk()) {
-    status = service->Start();
+    for (auto& grpc_eps : grpc_endpoint_services_) {
+      if (grpc_eps != nullptr) {
+        status = grpc_eps->Start();
+      }
+    }
   }
 
   if (!status.IsOk()) {
-    service.reset();
+    for (auto& grpc_eps : grpc_endpoint_services_) {
+      if (grpc_eps != nullptr) {
+        grpc_eps.reset();
+      }
+    }
   }
 
-  return std::move(service);
+  return status;
 }
 
-std::unique_ptr<nvidia::inferenceserver::HTTPServer>
-StartHttpService(nvidia::inferenceserver::InferenceServer* server)
+nvidia::inferenceserver::Status
+StartMultipleHttpService(
+    nvidia::inferenceserver::InferenceServer* server,
+    std::map<int32_t, std::vector<std::string>>& port_map)
 {
-  std::unique_ptr<nvidia::inferenceserver::HTTPServer> service;
   nvidia::inferenceserver::Status status =
       nvidia::inferenceserver::HTTPServer::Create(
-          server, http_port_, http_thread_cnt_, &service);
+          server, port_map, http_thread_cnt_, &http_endpoint_services_);
   if (status.IsOk()) {
-    status = service->Start();
+    for (auto& http_eps : http_endpoint_services_) {
+      if (http_eps != nullptr) {
+        status = http_eps->Start();
+      }
+    }
   }
 
   if (!status.IsOk()) {
-    service.reset();
+    for (auto& http_eps : http_endpoint_services_) {
+      if (http_eps != nullptr) {
+        http_eps.reset();
+      }
+    }
   }
 
-  return std::move(service);
+  return status;
 }
 
 bool
 StartEndpoints(nvidia::inferenceserver::InferenceServer* server)
 {
+  size_t i;
+  std::map<int32_t, std::vector<std::string>> port_map;
+  nvidia::inferenceserver::Status create_status;
   LOG_INFO << "Starting endpoints, '" << server->Id() << "' listening on";
 
-  // Enable gRPC endpoint if requested...
-  if (grpc_port_ != -1) {
-    LOG_INFO << " localhost:" << std::to_string(grpc_port_)
-             << " for gRPC requests";
-    grpc_service_ = StartGrpcService(server);
-    if (grpc_service_ == nullptr) {
+  // Enable gRPC endpoints if requested...
+  if (allow_grpc) {
+    std::vector<std::string> endpoint_names = {"status", "health", "profile",
+                                               "infer"};
+    std::vector<int32_t> endpoint_ports = {grpc_status_port_, grpc_health_port_,
+                                           grpc_profile_port_,
+                                           grpc_infer_port_};
+
+    // Group by port numbers
+    for (i = 0; i < endpoint_ports.size(); i++) {
+      if (endpoint_ports[i] != -1) {
+        port_map[endpoint_ports[i]].push_back(endpoint_names[i]);
+      }
+    }
+
+    for (i = 0; i < port_map.size(); i++) {
```

Comment:
What is this for?

Reply:
No longer needed since this is done in grpc/http_server.cc now